GW2EIEvtcParser/ParsedData/CombatData.cs

```diff
@@ -1,1746 +1,1409 @@
+                return true;
             }
-            return new List<BreakbarDamageEvent>();
         }
+        effectEvents = null;
+        return false;
+    }
 
-        /// <summary>
-        /// Returns list of crowd control events taken by Agent
-        /// </summary>
-        /// <param name="dst"></param> Agent
-        /// <returns></returns>
-        public IReadOnlyList<CrowdControlEvent> GetIncomingCrowdControlData(AgentItem dst)
+    /// <returns>true on success</returns>
+    public bool TryGetEffectEventsByGUIDs(Span<GUID> effects, out List<EffectEvent> effectEvents)
+    {
+        //TODO(Rennorb) @perf: fid average complexity
+        effectEvents = new(effects.Length * 10);
+        foreach (var effectGUID in effects)
         {
-            if (_crowControlTakenData.TryGetValue(dst, out List<CrowdControlEvent> res))
+            if (TryGetEffectEventsByGUID(effectGUID, out var events))
             {
-                return res;
+                effectEvents.AddRange(events);
             }
-            return new List<CrowdControlEvent>();
         }
 
-        public IReadOnlyList<AbstractMovementEvent> GetMovementData(AgentItem src)
-        {
-            if (_statusEvents.MovementEvents.TryGetValue(src, out List<AbstractMovementEvent> res))
-            {
-                return res;
-            }
-            return new List<AbstractMovementEvent>();
-        }
+        return effectEvents.Count > 0;
+    }
 
-        public IReadOnlyList<EffectEvent> GetEffectEventsBySrc(AgentItem src)
+    /// <summary>
+    /// Returns effect events by the given agent and effect GUID.
+    /// </summary>
+    /// <returns>true on found effect with entries > 0</returns>
+    public bool TryGetEffectEventsBySrcWithGUID(AgentItem agent, GUID effect, [NotNullWhen(true)] out List<EffectEvent>? effectEvents)
+    {
+        if (TryGetEffectEventsByGUID(effect, out var effects))
         {
-            if (_statusEvents.EffectEventsBySrc.TryGetValue(src, out List<EffectEvent> list))
+            var result = effects.Where(effect => effect.Src == agent).ToList();
+            if (result.Count > 0)
             {
-                return list;
+                effectEvents = result;
+                return true;
             }
-            return new List<EffectEvent>();
         }
 
-        public IReadOnlyList<EffectEvent> GetEffectEventsByDst(AgentItem dst)
-        {
-            if (_statusEvents.EffectEventsByDst.TryGetValue(dst, out List<EffectEvent> list))
-            {
-                return list;
-            }
-            return new List<EffectEvent>();
-        }
+        effectEvents = null;
+        return false;
+    }
 
-        public IReadOnlyList<EffectEvent> GetEffectEventsByEffectID(long effectID)
+    /// <summary>
+    /// Appends effect events by the given agent and effect GUID.
+    /// </summary>
+    public void AppendEffectEventsBySrcWithGUID(AgentItem agent, GUID effect, List<EffectEvent> effectEvents)
```

Comment:
Why are the Append methods public?

Reply:
To be completely honest the reason is because i don't believe in the OOP paradigm of encapsulation and therefore declare almost everything public in my own work. I tried doing a reasonable job with this, but that one must have slipped through. We can certainly make it private if that is preferable.

Reply:
Ok, so after having a second look at this:
Why shouldn't the append methods be public? The TryGet methods that currently use them are are just wrappers around a loop of append calls, so why shouldn't they share the same visibility? Sometimes one or the other method might be more convenient to call, why take away that choice from the consumer and pretend to know which one will be more usefull?

This is also the main reason why I have such issues with encapsulation: It assumes you as the library author know better what the consumer wants than the consumer.

Reply:
That is quite the tangent, "To be completely honest the reason is because i don't believe in the OOP paradigm of encapsulation and therefore declare almost everything public in my own work" was a perfectly acceptable answer =D. The API does not expose something that is private nor allow something private to be modified. I saw that the method was only used locally, thus mu curiosity.

Reply:
I will have another look and find out if it would be sensible to use the append version in other places aswell, then report back before we resolve this.

Reply:
Alright, looks like its not really feasible to propagate the append methods even more, but they might come in handy at some point down the line.